### PR TITLE
Implementing a Generic Singly linked list Data Structure in Java

### LINKED ISSUE #4382: Implementing a Generic, singly Linked List data structure in Java
#### Proposed title of article
Implementing a Generic, singly Linked List data structure in Java
#### Proposed article introduction
Linear Data Structures that store values in nodes are known as Linked Lists. As we know, each Node has two properties: the value of the node and, if present, a connection to the next node. Linked Lists can be of any data type, including String, Boolean, Float, Character, and so on. 
We will be creating a generic Linked List Data Type that can store data of any type.

#### Key takeaways 
- Primary member functions of a linked list
- Creating a generic linked list
- What is a generic linked list and how we implement them

#### References
N/A
### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by LynnMumo**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4382

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by LynnMumo** [comment]
@lalith1403 @Dawe-7 
This is the new PR
Duplicate of #4691

File: content/articles/implementing-a-generic-singly-linked-list-data-structure-in-java/index.md [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+        // head-pointing dummy node
+        prev.next = head;
+        // The node after the current node that points forward
+        node<T> next = head.next;
+```
+
+Take advantage of Generics' SinglyLinkedList. `Node<T>` will be the type of node variable. Now that we have covered the basics, let's get down to business by implementing the methods below:
+- Last but not least, we'll use add(T data).
+- Add the element at the top of the list.
+- Add an element to a specific location, use add(T data, int index).
```
Comment (Dawe-7) [inline content/articles/implementing-a-generic-singly-linked-list-data-structure-in-java/index.md:237]: @LynnMumo, The paragraph here doesn't seem to flow very, especially how you started listing the methods "Last but not least. Kindly rephrase it in a more comprehensive way.

**Review by Dawe-7 — CHANGES_REQUESTED** [review]
Hello @LynnMumo, could you check on the comment i've left.

**Comment by hectorkambow** [comment]
@Dawe-7 we will be closing this PR to allow the author the proper time to make any and all needed corrections.

This is a duplicate PR - if any more changes were needed (in general) we should close the PR immediately especially if its a duplicate PR (generally there are always exceptions).

 cc @LynnMumo 
 
 It seems this PR is still quite a bit aways from being completed.  As a rule of thumb - please only submit articles (pull requests) with a complete and fully polished article - ready to be published. 
 
All revisions and edits should be completed on your own forked repo (as to not take up room in the queue). 
 
We will be closing this and please re-submit when it is _ready to be published_ as a new PR (and you can add the duplicate label) and ping the reviewer and myself. 

Closing a PR does NOT mean the article can not be published (or fixed) it simply means that the edits, revisions, and fixes will happen outside of the queue.

**Comment by Dawe-7** [comment]
> @Dawe-7 we will be closing this PR to allow the author the proper time to make any and all needed corrections.
> 
> This is a duplicate PR - if any more changes were needed (in general) we should close the PR immediately especially if its a duplicate PR (generally there are always exceptions).
> 
> cc @LynnMumo
> 
> It seems this PR is still quite a bit aways from being completed. As a rule of thumb - please only submit articles (pull requests) with a complete and fully polished article - ready to be published.
> 
> All revisions and edits should be completed on your own forked repo (as to not take up room in the queue).
> 
> We will be closing this and please re-submit when it is _ready to be published_ as a new PR (and you can add the duplicate label) and ping the reviewer and myself.
> 
> Closing a PR does NOT mean the article can not be published (or fixed) it simply means that the edits, revisions, and fixes will happen outside of the queue.

Okay @hectorkambow

**Comment by LynnMumo** [comment]
@Dawe-7 
I have done some edits for the requested changes. You can reopen the PR and have a look at my last commit. Thanks.
cc @hectorkambow

**Comment by Dawe-7** [comment]
@linetachieng, reopening so changes can be effected.

**Comment by hectorkambow** [comment]
@LynnMumo 

Good afternoon and thank you for submitting your topic to the EngEd program. 
If it seems this PR is still quite a bit aways from being completed. 
As a rule of thumb - please only submit articles (pull requests) with a complete and fully polished article - ready to be published. 
All revisions and edits should be completed on your own forked repo (as to not build up a backlog in the queue). 
We will be closing this and please re-submit when it is ready to be published as a new PR (and you can add the duplicate label) and ping the reviewer and myself. 

Closing a PR does NOT mean the article can not be published (or fixed) it simply means that the edits, revisions, and fixes will happen outside of the queue.
 
 
👍

**Comment by LynnMumo** [comment]
@hectorkambow the article was ready. I did some changes when the article was closed. I don't understand why you closed it again.

**Comment by Dawe-7** [comment]
@hectorkambow, the author created a new PR and it was in final review on my end. I was going through everything to ensure it was ready for publishing. The article did not have extensive revisions, any delays caused were on my end.

**Comment by Dawe-7** [comment]
@hectorkambow, reopening to finish review.

**Review by Dawe-7 — APPROVED** [review]
@LynnMumo, kindly go through the changes made throughout the review process so as to improve on the flow and format of your future articles. Thank you.

@hectorkambow , this article is ready for final review.

**Comment by hectorkambow** [comment]
I'll be jumping on this one

**Comment by hectorkambow** [comment]
@LynnMumo 

Question how did you start to submit your article without signing up with the program?

**Comment by LynnMumo** [comment]
We were having a discussion about EngEd with my friends and that's how I
found out that I never enrolled. All I knew was that you only submit a
topic and the other activities follow. I'm sorry.

**Pull request merged**